subsys/bluetooth/mesh/proxy_msg.c

```diff
@@ -243,7 +190,70 @@ int bt_mesh_proxy_msg_send(struct bt_mesh_proxy_role *role, uint8_t type,
 	return 0;
 }
 
-void bt_mesh_proxy_msg_init(struct bt_mesh_proxy_role *role)
+static void proxy_msg_init(struct bt_mesh_proxy_role *role)
 {
+	uint8_t i, len;
+	uint8_t *buf;
+
+	/* Check if buf has been allocated, in this way, we no longer need
+	 * to repeat the operation.
+	 */
+	if (role->buf.__buf) {
+		net_buf_simple_reset(&role->buf);
+		return;
+	}
+
+	i = bt_conn_index(role->conn);
+	if (!i) {
+		len = PROXY_MSG_FIRST_BUF_LEN;
+		buf = bufs;
+	} else {
+		len = PROXY_BUF_LEN_MAX;
+		buf = &bufs[PROXY_MSG_FIRST_BUF_LEN + (PROXY_BUF_LEN_MAX * (i - 1))];
+	}
+
+	net_buf_simple_init_with_data(&role->buf, buf, len);
+
+	net_buf_simple_reset(&role->buf);
+
 	k_work_init_delayable(&role->sar_timer, proxy_sar_timeout);
 }
+
+struct bt_mesh_proxy_role *bt_mesh_proxy_role_setup(struct bt_conn *conn,
+						    proxy_send_cb_t send,
+						    proxy_recv_cb_t recv)
+{
+	struct bt_mesh_proxy_role *role;
+
+	role = &roles[bt_conn_index(conn)];
```

Comment:
The buffer sizes assume that we'll only use `role[0]` when running PB-GATT, but if there already were another connection open when the PB-GATT connection was established, it'll be `1` here. What will happen to the buffer in this case?

Reply:
Isn't the connection using pb-gatt 0?

Reply:
Do you mean that mesh may coexist with other samples?

Reply:
If this is the case, then I have to define all of them as 66 bytes, which is consistent with now.

Reply:
I'm fine with all connection use same buffer len, and it's true for pb-gatt client.
So, add this now is OK.